packages/data-table-server/src/dataTableService/internal/EntitiesDataTableService.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { TupaiaApiClient } from '@tupaia/api-client';
+import { yup } from '@tupaia/utils';
+import { DataTableService } from '../DataTableService';
+import { yupSchemaToDataTableParams } from '../utils';
+
+const paramsSchema = yup.object().shape({
+  hierarchy: yup.string().default('explore'),
+  entityCodes: yup.array().of(yup.string().required()).required(),
+  filter: yup.object(),
+  fields: yup.array().of(yup.string().required()).default(['code']),
+  includeDescendants: yup.boolean().default(false),
+});
+
+const configSchema = yup.object();
+
+type Entity = Record<string, unknown>;
+
+/**
+ * DataTableService for pulling data entityApi.getEntities() and entityApi.getDescendantsOfEntities()
+ */
+export class EntitiesDataTableService extends DataTableService<
+  typeof paramsSchema,
+  typeof configSchema,
+  Entity
+> {
+  public constructor(apiClient: TupaiaApiClient, config: unknown) {
+    super(paramsSchema, configSchema, apiClient, config);
+  }
+
+  protected async pullData(params: {
+    hierarchy: string;
+    entityCodes: string[];
+    filter?: Record<string, unknown>;
+    fields: string[];
+    includeDescendants: boolean;
+  }) {
+    const { hierarchy, entityCodes, filter, fields, includeDescendants } = params;
+
+    const entities = await this.apiClient.entity.getEntities(hierarchy, entityCodes, {
+      fields,
+      filter,
+    });
+
+    if (!includeDescendants) {
```

Comment:
Thoughts on this `includeDescendants` flag? I wanted there to be an easy way for users to pull all the entities in a hierarchy, but you could argue we should make a separate `entity_descendants` data-table for that?

Reply:
I think we definitely want to sometimes not pull descendants. For instance, if we have a list of entity codes and we want to get their names, we want to use this data-table to fetch the names from the codes.

But we can always change this later if we don't like how it behaves 👍